apps/staging-enterprise/__tests__/cli

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+set -e
+
+cd "`dirname "$0"`/.."
+
+CLI="../../cli/dist/grouparoo.js"
+
+printf "*** Starting Tests ***\r\n"
+
+OUTPUT_HELP=`$CLI --help`
+printf "help flag"
+if [[ "$OUTPUT_HELP" == *"Learn more at www.grouparoo.com/docs"*  ]]; then
+    printf " ✅\r\n"
+else
+    printf " 🚫\r\n"
+    exit 1
+fi
+
+OUTPUT_VALIDATE=`$CLI validate`
+printf "validate"
+if [[ "$OUTPUT_VALIDATE" == *"Validation succeeded"*  ]]; then
+    printf " ✅\r\n"
+else
+    printf " 🚫\r\n"
+    exit 1
+fi
+
+OUTPUT_APPLY=`$CLI apply`
+printf "apply"
+if [[ "$OUTPUT_APPLY" == *"Config applied"*  ]]; then
+    printf " ✅\r\n"
+else
+    printf " 🚫\r\n"
+    exit 1
+fi
+
+printf "*** Testing Generators ***\r\n"
+
+TEST_GENERATOR(){
+  OUTPUT_GENERATE=`$CLI generate $i $i --parent foo`
+  if [[ "$OUTPUT_GENERATE" == *"wrote /"*  ]]; then
+      printf " - generate $i ✅\r\n"
+  else
+      printf " - generate $i 🚫\r\n"
+      exit 1
+  fi
+
+  # cleanup
+  FILENAME=`echo "$OUTPUT_GENERATE" | grep wrote | awk '{print $3}'`
+  rm $FILENAME
+}
+
+GENERATE_TEMPLATES=`$CLI generate --list | grep "(id" | awk '{print $1}' | perl -pe 's/\e\[[0-9;]*m(?:\e\[K)?//g'` # the perl part is to remove the underline and bold escape chars
+for i in $GENERATE_TEMPLATES; do
+    TEST_GENERATOR $i & # The "&" backgrounds the function
```

Comment:
This is basically a fork bomb... but... it's fine.

Reply:
It turns out it wasn't fine after all! https://github.com/grouparoo/grouparoo/pull/1463/commits/bed284d6f6666d5b0560b73a84747bdd45acb713 is an awesome solution